File: AIO/AIO 2024/[p2] Subbookkeeper.py
input_file = open('bookin.txt', 'r')
N = int(input_file.readline())
word = input_file.readline()
 
output = 0
current_letter = ''
score = 0
 
for i in range(len(word)):
    #print(current_letter, word[i], score, output)
    if word[i] == current_letter or word[i] == '?':
        score += 1
    else:
        output += score
        score = 0
        current_letter = word[i]
output += score
 
output_file = open('bookout.txt', 'w')
output_file.write(str(output))
output_file.close()

File: AIO/AIO 2024/[p2] Subbookkeeper.cpp
#include <bits/stdc++.h>
using namespace std;
 
signed main() noexcept {
    int N;
    cin >> N;
    string word;
    cin >> word;
 
    int output = 0;
    char current_letter = ' ';
    int score = 0;
 
    for(char letter : word) {
        if(letter == current_letter or letter == '?') {
            score++;
        } else {
            output += score;
            score = 0;
            current_letter = letter;
        }
    }
    output += score;
 
    cout << output;
}
